python/pyarrow/tests/test_extension_type.py

```diff
@@ -1079,3 +1082,279 @@ def test_array_constructor_from_pandas():
         pd.Series([1, 2, 3], dtype="category"), type=IntegerType()
     )
     assert result.equals(expected)
+
+
+class TensorType(pa.ExtensionType):
```

Comment:
Use `FixedShapeTensorType`, to follow the name change?

And the same for the TensorArray